src/legacy/core_plugins/embeddable_api/public/actions/apply_filter_action.ts

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
```

Comment:
if i understand correctly this is used by saved search only ? visualize currently directly applies filters and in the future it will need a custom apply filter action. can we move this out of the api folder ?

Reply:
My intention was to have this used by visualizations as well, but since your filter expression exploration, we'll have to discuss more.  I still feel like a filter trigger is cross embeddable.  Any interaction with an embeddable that results in an array of filters (kibana filters), or field: value pairs, could use this trigger.

I'm starting to think that it makes sense to determine how many triggers to use by the trigger context they emit. It will make deciding whether attaching an action to a given trigger makes sense or not easier.  For instance, drill down links... any action that produces a filter seems like a good candidate for a saved object drill down link.  Or an API call where the data point clicked on could be useful information.  If you have an action that doesn't use that data, it probably fits better in the context menu, where there is no trigger context information.  

Maybe we can also think of emitting multiple triggers at the same time.  Consider clicking on an ip address in a data table.  All filter actions could be applicable, but in addition specialized actions that are only useful when dealing with an ip address field (like look up domain information, search in SIEM, search in logs, etc).

So I don't think it makes sense to a trigger where the context is a global concept - Kibana Filters (and eventually maybe more user friendly facing data, like field name, field value), to be tied to saved searches only.  All embeddables should be able to use this so if a developer attaches a drill down link action to it, it shows up on all embeddables.

tl;dr, I think this is debatable and worth more discussion, but since I'm not sold on it being in searches, I would prefer to leave where it is and kick to a separate discussion for where it's final resting place should be.